src/rules/no-stale-state-across-await.ts

```
@@ -0,0 +1,182 @@
+          // Recursively walk child nodes
+          for (const key in n) {
+            if (key === 'parent') continue;
+            const value = (n as any)[key];
+            if (Array.isArray(value)) {
+              for (const item of value) {
+                if (item && typeof item === 'object' && item.type) {
+                  // Skip nested functions unless we're explicitly walking into callbacks
+                  if (
+                    skipNestedFunctions &&
+                    (item.type === AST_NODE_TYPES.FunctionDeclaration ||
+                     item.type === AST_NODE_TYPES.FunctionExpression ||
```

ESLint

[error] Insert `·`

(prettier/prettier)